src/decimal.rs

```diff
@@ -124,6 +128,10 @@ pub struct UnpackedDecimal {
     archive_attr(derive(Clone, Copy, Debug))
 )]
 #[cfg_attr(feature = "rkyv-safe", archive_attr(derive(CheckBytes)))]
+#[cfg_attr(
+feature = "scale-codec",
+derive(Decode, Encode, TypeInfo, MaxEncodedLen),
```

Comment:
Side note (no change required): these `derive` attributes are causing a bit of code bloat. I want to rethink how we do this in v2. It may be by exposing a wrapper `struct` and aliasing it however all in all, this current approach isn't entirely scalable.